grafana_import/grafana.py

```diff
@@ -305,18 +308,23 @@ def import_dashboard(self, dashboard: t.Dict[str, t.Any]) -> bool:
                     )
             # ** case d) send a copy to existing dash : update existing
             elif new_dash["folderId"] == old_dash["folderId"]:
-                if "uid" not in new_dash["dashboard"] or new_dash["dashboard"]["uid"] != old_dash["uid"]:
+                if ("uid" not in new_dash["dashboard"]
+                        or new_dash["dashboard"]["uid"] != old_dash["uid"]
+                        or new_dash["dashboard"]["id"] != old_dash["id"]):
                     if self.overwrite:
-                        new_dash["dashboard"]["uid"] = old_dash["uid"]
+                        if not self.keep_uid:
+                            new_dash["dashboard"]["uid"] = old_dash["uid"]
```

Comment:
Is this fragment correct, or should the check be inverted?
```suggestion
                        if self.keep_uid:
                            new_dash["dashboard"]["uid"] = old_dash["uid"]
```

Reply:
@amotl , `-keep_uid` argument is used to preserve uid provided in dashboard JSON file under any conditions. The line `new_dash["dashboard"]["uid"] = old_dash["uid"]` overwrites JSON uid by the existing dashboard uid  So, to keep the JSON uid, when we provide `-keep-uid` arg even with `-overwrite` arg, we need to set:
`if not self.keep_uid:
        new_dash["dashboard"]["uid"] = old_dash["uid"]`
 Hope I expressed that clearly :). If not, you are welcome to ask more questions.